src/features/calendar.ts

```diff
@@ -0,0 +1,110 @@
+import bot from '../bot'
+import { Telegraf, Context } from 'telegraf'
+import { callbackQuery } from "telegraf/filters"
+import moment from "moment";
+
+bot.command('date', (ctx: Context) => {
+	ctx.reply('Pick a year:', {
+	reply_markup: {
+    	inline_keyboard: generateCalendarKeyboardAnio(),
+    },
+  })
+})
+bot.on(callbackQuery("data"), ctx => {
+    let [actionType, actionValue,days] = ctx.callbackQuery.data.split(':')
+	let date
+    switch (actionType) {
+    	case 'anio':
+        	ctx.answerCbQuery()
+        	ctx.editMessageText('Pick a month:', {
+            	reply_markup: {
+                	inline_keyboard: generateCalendarKeyboardMonth(actionValue),
+                },
+              })
+          break
+        case 'month':
+            ctx.answerCbQuery()
+            ctx.editMessageText('Pick a day:', {
+            	reply_markup: {
+            		inline_keyboard: generateCalendarKeyboardDay(actionValue,days),
+              	},
+            })
+            
+          break
+        case 'day':
+			date = actionValue
+        	ctx.editMessageText(`Date: ${date}`)
+            ctx.answerCbQuery()
+		break
+    }
+
+  })
+function getDate(){
+	bot.on(callbackQuery("data"), ctx => {
+	ctx.reply('Pick a year:', {
+		reply_markup: {
+			inline_keyboard: generateCalendarKeyboardAnio(),
+		},
+	  })
+	})
+}
+function generateCalendarKeyboardAnio() {
+    const keyboard = []
+    const anios = [
+      { name: '2023', anio: 2023 },
+      { name: '2024', anio: 2024 },
+    ]
+    for (let i = 0; i < anios.length; i += 2) {
+    	const row = []
+        const anio1 = anios[i]
+        const anio2 = i + 1 < anios.length ? anios[i + 1] : null
+        row.push({ text: anio1.name, callback_data: `anio:${anio1.anio}` })
+        if (anio2) {
+        	row.push({ text: anio2.name, callback_data: `anio:${anio2.anio}` })
+        }
+        keyboard.push(row)
+    }
+    return keyboard
+  }
+
+function generateCalendarKeyboardMonth(anio: any) {
+	const keyboard = []
+	let days
+	const months = [
+    	{ name: 'January'},{ name: 'February'},{ name: 'March' },{ name: 'April'},{ name: 'May'},{ name: 'June' },
```

Comment:
Constants have to be declared in constants file.
If you don't need other attributes in 'month' object, better use only strings instead objects.